python/ray/tests/kuberay/test_autoscaling_e2e.py

```diff
@@ -0,0 +1,232 @@
+import logging
+import os
+import pathlib
+import tempfile
+import unittest
+import subprocess
+
+from typing import Any, Dict
+
+import yaml
+
+from ray.tests.kuberay.utils import (
+    get_pod,
+    kubectl_exec,
+    wait_for_pods,
+    wait_for_pod_to_start,
+    wait_for_ray_health,
+    wait_for_crd,
+)
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(
+    level=logging.INFO,
+    format="[%(levelname)s %(asctime)s] " "%(filename)s: %(lineno)d  " "%(message)s",
+)
+
+# This image will be used for both the Ray nodes and the autoscaler.
+# The CI should pass an image built from the test branch.
+RAY_IMAGE = os.environ.get("RAY_IMAGE", "rayproject/ray:413fe0")
+logger.info(f"Using image {RAY_IMAGE} for autoscaler and Ray nodes.")
+# The default "rayproject/ray:413fe0" is the currently pinned autoscaler image
+# (to be replaced with rayproject/ray:1.12.0 upon 1.12.0 release).
+
+# Parent directory of Ray repository
+RAY_PARENT = str(pathlib.Path(__file__).resolve().parents[5])
+# Path to example config rel RAY_PARENT
+EXAMPLE_CLUSTER_PATH = "ray/python/ray/autoscaler/kuberay/ray-cluster.complete.yaml"
```

Comment:
Do you think this yml would be too heavy for a docker kind cluster running inside a vm?

Reply:
This test spins up a head node and two workers, each requesting a CPU each.
Altogether, the test fits into the VM's 4 CPUs (but just barely).

From my experiments so far, it passes pretty consistently!

Reply:
What about memory allocation?

Reply:
Added!

Reply:
One advantage of using this yaml is that it is the same that is used in the documentation, so it will make sure that keeps working :)

Reply:
yyyep, that's the idea